compiler/rustc_borrowck/src/path_utils.rs

```diff
@@ -166,7 +165,7 @@ pub(crate) fn is_upvar_field_projection<'tcx>(
         Some((place_base, ProjectionElem::Field(field, _ty))) => {
             let base_ty = place_base.ty(body, tcx).ty;
             if (base_ty.is_closure() || base_ty.is_coroutine())
-                && (!by_ref || upvars[field.index()].by_ref)
+                && (!by_ref || upvars[field.index()].info.capture_kind.is_by_ref())
```

Comment:
I've seen this line multiple times. Does it deserve a helper method on CapturedPlace?